### DISCUSSION TITLE
Is that possible to run two versions of tensorflow by tox?

### DISCUSSION THREAD

**Discussion by Wapiti08** [tox-dev/tox#2155] [Q&A]
Describe what improvement you want and how would this be used. Thanks!

**Comment by gaborbernat** [comment]
Can you please clarify what exactly you want to achieve?

**Comment by Wapiti08** [comment]
I am struggling to test two versions of tensorflows in a project. I tried to use virtualenv to activate and deactivate the environment to achieve that. I want to know whether tox can provide more convenient way to do that part.

**Reply by gaborbernat** [reply]
Yes it can.

**Reply by Wapiti08** [reply]
Okay. Thanks for your response. I have a problem as well to mention. Sorry I am newbie for this tool.
I have the following code:
```
[tox]
envlist = test_service
skipdist = True

[testenv]
install_command = pip install {opts} {packages}
basepython = 3.8.1.final.0

[testenv:test_service]
deps =
  -rrequirements.txt

setenv=
    PYTHONPATH=.

commands=
    sudo /bin/bash install_script.sh

```
When I test is with tox -e xx. I always got the error about missing project.toml. I am not sure how to fix that part. Can you please give me some advice? Thanks a lot.

**Reply by gaborbernat** [reply]
Generally whenever you're requesting help you need to provide the tox output, something like: `tox --version` and `tox -vve xx`. I have a few more questions:

- why define test_service but then invoke xx
- why set PYTHONPATH = .
- why do `sudo /bin/bash install_script.sh`
- why set `basepython = 3.8.1.final.0`?

**Comment by Wapiti08** [comment]
Here is the output, sorry about that:
```
tox -vve test_service
using tox.ini: xxxtox.ini (pid 44872)
using tox-3.24.1 from c:\users\ren\appdata\local\programs\python\python38\lib\site-packages\tox\__init__.py (pid 44872)
ERROR: No pyproject.toml or setup.py file found. The expected locations are:
  E:\xxx\pyproject.toml or E:\xxx\setup.py
You can
  1. Create one:
     https://tox.readthedocs.io/en/latest/example/package.html
  2. Configure tox to avoid running sdist:
     https://tox.readthedocs.io/en/latest/example/general.html
  3. Configure tox to use an isolated_build
```

The goal I want to achieve it to test the install process for requirements.txt. The version of python is the testing python version I used on my system. I think I need to read the instruction more about how to use that tool.
